ddtrace/internal/writer.py

```diff
@@ -64,19 +67,39 @@ def flush_queue(self):
             log.error('error while filtering traces: {0}'.format(err))
             return
 
-        if not traces:
-            return
+        traces_flushed = len(traces)
+        traces_flushed_error = 0
 
         # If we have data, let's try to send it.
         traces_responses = self.api.send_traces(traces)
         for response in traces_responses:
             if isinstance(response, Exception) or response.status >= 400:
                 self._log_error_status(response)
+                traces_flushed_error += 1
             elif self._priority_sampler:
                 result_traces_json = response.get_json()
                 if result_traces_json and 'rate_by_service' in result_traces_json:
                     self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
 
+        # Dump statistics
+        # NOTE: Do not use the buffering of dogstatsd as it's not thread-safe
+        # https://github.com/DataDog/datadogpy/issues/439
+        if self.dogstatsd:
+            for status, grouped_responses in itertools.groupby(
+                    sorted(traces_responses, key=lambda r: r.status),
+                    key=lambda r: r.status):
+                self.dogstatsd.increment('datadog.tracer.traces.api.status.%d' % status, len(list(grouped_responses)))
+            if hasattr(time, 'thread_time_ns'):
+                self.dogstatsd.increment('datadog.tracer.writer.cpu_time', time.thread_time_ns())
+            self.dogstatsd.increment('datadog.tracer.traces.flushed', traces_flushed)
```

Comment:
I think this and the following might need to be named differently.

This is telling us how many traces are being flushed, but the `.errors` one is how many payloads/api calls failed.

we might want to add the number of payloads sent.

Would be nice to know that 100 traces are being flushed, but resulted in 10 payloads being sent

Reply:
Hm, you actually have the number of flush going out if look at the number of `datadog.tracer.traces.api.status.200`, no?

Reply:
True, but we don't have a total count.

We should consider something like:

`datadog.tracer.traces.api`

along with

`datadog.tracer.traces.api.by_status`

Having both. Mostly because if you want a total of API calls made regardless of status code, then you need to basically do:

`datadog.tracer.traces.api.200 + datadog.tracer.traces.api.404 + datadog.tracer.traces.api.403 + datadog.tracer.traces.api.503`, etc etc, and better hope you didn't miss one 😆

Reply:
also could do like `api.requests` vs `api.responses.by_status`, just semantically if that makes sense or not? (btw, I am terrible at naming things)

Reply:
I hope we don't have a maximum length on metric names 😆  Other than that, that sounds good, added all of that.

Reply:
there is a max length, but these aren't too bad